test/unit/full_text_search/attachment_test.rb

```diff
@@ -104,7 +104,12 @@ def teardown_server
     def fixture_file_path(name)
       dir = File.dirname(__FILE__)
       path = Pathname(File.join(dir, "..", "..", "files", name)).expand_path
-      fixture_path = Pathname(self.class.fixture_path)
+      fixture_path = if self.class.respond_to?(:fixture_paths=)
+                       # For Rails >= 7.1
+                       Pathname(self.class.fixture_paths.first)
```

Comment:
I think this is not a good way. But at the same time, I should think about whether we should support the multi-fixture paths situation in this test or not.

Reply:
Does this work?

```diff
diff --git a/test/unit/full_text_search/attachment_test.rb b/test/unit/full_text_search/attachment_test.rb
index 7fad7fb..8467279 100644
--- a/test/unit/full_text_search/attachment_test.rb
+++ b/test/unit/full_text_search/attachment_test.rb
@@ -104,9 +104,21 @@ module FullTextSearch
     def fixture_file_path(name)
       dir = File.dirname(__FILE__)
       path = Pathname(File.join(dir, "..", "..", "files", name)).expand_path
-      fixture_path = Pathname(self.class.fixture_path)
-      fixture_path += "files" if Rails::VERSION::MAJOR >= 6
-      path.relative_path_from(fixture_path)
+      if self.class.respond_to?(:fixture_paths)
+        fixture_paths = self.class.fixture_paths.collect do |fixture_path|
+          Pathname(fixture_path)
+        end
+      else
+        fixture_paths = [Pathname(self.class.fixture_path)]
+      end
+      fixture_paths = fixture_paths.collect do |fixture_path|
+        fixture_path += "files" if Rails::VERSION::MAJOR >= 6
+        path.relative_path_from(fixture_path)
+      end
+      fixture_paths.each do |fixture_path|
+        return fixture_path if File.exist?(fixture_path)
+      end
+      raise InvalidArgument, "nonexistent fixture: #{name.inspect}"
     end
 
     def capture_log
```

Reply:
Thanks for your suggestion. This code doesn't work because of the following point.
```ruby
fixture_paths.each do |fixture_path|
  return fixture_path if File.exist?(fixture_path) # <- Here it doesn't work because the relative path is from the specific fixture_path, not from this file.
end
```

On the other hand, From Rails v7, I just realized that we don't have to pass a path to `fixture_file_upload` relative to `fixture_path` anymore. So we can just pass the absolute path.
fix: [0aaae39](https://github.com/clear-code/redmine_full_text_search/pull/133/commits/0aaae39b2bf8797472608bd5a864bfa7932199b1)
- https://github.com/rails/rails/blob/main/guides/source/7_0_release_notes.md#removals-2
- https://github.com/rails/rails/commit/1f4a5414219fbc2052c54d11447770e1269c7d1f